### ISSUE TITLE
New SIDs, Fixes, and other minor problems for VHHH

### ISSUE THREAD

**Issue by siyuanr** [zlsa/atc#370]
According to the charts [published by the HKAIS](http://www.ais.gov.hk/HK_AIP/ad.htm) (a government agency), we are missing some SIDs and important fixes.

Fixes that needs to be added for new SIDs:

``` json
"ATENA": ["N22d24m39.85", "E114d23m11.88"],
"COLEY": ["N22d06m41.03", "E114d18m37.63"],
"RAMEN": ["N22d09m39.22", "E114d05m09.89"],
"PORPA": ["N22d20m09.10", "E114d01m16.30"],
"ROVER": ["N22d20m35.58", "E114d01m39.12"]
```

Could someone add those to `vhhh.json` even if what's described below couldn't be done?

Now comes the tricky part, which is in some way an extension of Issue #324 in `VHHH`. However, in Hong Kong's case, paths with the same end fix but different departing runways are treated as different SIDs, which are given different names, and there may be more than one SID between a runway and an end fix.

For example, for the same `BEKOL` SID, the HKAIS has, for RWYs 7L, 7R, 25L, 25R, respectively:

``` json
"BEKOL 3C": ["ROVER", "TD", "ATENA", "BEKOL", "IDUMA"],
"BEKOL 3A": ["PORPA", "TD", "ATENA", "BEKOL", "IDUMA"],
"BEKOL 2B": ["PRAWN", "RUMSY", "TUNNA", "SAMON", "BEKOL", "IDUMA"],
"BEKOL 2D": ["PRAWN", "RUMSY", "TUNNA", "SAMON", "BEKOL", "IDUMA"]
```

Will this cause any problems? If not, we could then proceed to write the different versions of the other SIDs.

We may also consider adding `ATHENA` SIDs for RWYs 7L, 7R, since the HKIAS has:

``` json
"ATENA 2C": ["ROVER", "RAMEN", "TD", "ATENA"],
"ATENA 2A": ["PORPA", "RAMEN", "TD", "ATENA"]
```

However, since `ATHENA` is still far away from the radar boundary, we may need to come up with an extender fix further to the east, just as what `IDUMA` is for `BEKOL`. Also the number of destinations may need to be expanded.

Another thing that will bring `VHHH` closer to reality is the flight altitude limit. For some reason I cannot issue a command to get a flight pass FL100. Per Hong Kong Regulations for the `OCEAN` SID in the game, the plane has to pass `TROUT` at or above FL140; for `BEKOL` SID the plane needs to pass `BEKOL` at or above FL150. Is there a away to change that?

**Comment by Maverick283** [comment]
As for the missing fixes, they can be added right now.

At this state, SIDs can not be assigned to runways. #324 only provides a workaround for this special case. Assigning SIDs to runways was somewhat postponed I think. It could be worth opening a separate issue for that.

As for the FL100 restriction, that is currently discussed as well. It was introduced in #344 when Flightlevels were brought to the input syntax. At least that's when I first noticed it, maybe it  wasnt @erikquinn who did this after all ;)

With the issues #339, #343, #356 and maybe #358 this might get resolved in one way or the other, and a climb above FL100 will be possible again.

**Comment by erikquinn** [comment]
#324 is my next project, and it will be based on runway, so this issue should be fixed within a week. Please don't "open a separate issue for that", but rather add comments to #324 if you have any additional details to share on how the SIDs should work. I'm not familiar with 3C/2A/2B/etc variants of a SID, so maybe you could provide some insight there, OR just link me to the actual SID charts and I should be able to figure it out. 

The restriction of 10,000ft js actually hard-coded, and should instead be based on the airport. That one is on the backlog, but is a quick fix so maybe I can get going on jt or at least create an issue that points someone else where to fix it. #344 just adds the "flight level" words to altitudes above the transition altitude; the 10,000' restriction has been there for as long as I can remember.

Thanks for all the info, @siyuanr, it'll be fixed sometime soon!

**Comment by siyuanr** [comment]
Just submitted a pull request to add new fixes under #372.

**Comment by siyuanr** [comment]
Just incorporated all my proposals in #372. These are changes specific to `vhhh.json`. To do so I referred to the latest official charts [here](http://www.ais.gov.hk/HK_AIP/ad.htm). In short:
- Fix `VH535` deleted because it is no longer used by [RNAV Approach 25R(z)](http://www.ais.gov.hk/HK_AIP/AIP/AD/HK_AD2-94G.pdf) or any other approaches;
- Fix `VH721` added since it is used in [RNAV approach 7R](http://www.ais.gov.hk/HK_AIP/AIP/AD/HK_AD2-91E.pdf);
- Fix `VH541` added to complement [RNAV approach 25L(y)](http://www.ais.gov.hk/HK_AIP/AIP/AD/HK_AD2-92G.pdf);
- Fix `SAMPU` added to complete the missed-approach route of [RNAV approach 7L](http://www.ais.gov.hk/HK_AIP/AIP/AD/HK_AD2-93G.pdf);
- Fixes `ATENA`, `BREAM`, `COLEY`, `PORPA`, `PORSH`, `ROBBE`, `ROVER`, `RAMEN`, `SHELY` added to construct new SIDs (once #324 is solved), among others [SID ATENA 1E/F](http://www.ais.gov.hk/HK_AIP/AIP/AD/HK_AD2-97%20ATENA%20EF.pdf), [SID LAKES 3A/C](http://www.ais.gov.hk/HK_AIP/AIP/AD/HK_AD2-97%20LAKES%20AC.pdf), and [SID TITAN 1E/F](http://www.ais.gov.hk/HK_AIP/AIP/AD/HK_AD2-97%20TITAN%20EF.pdf);
- Fix `SILVA` added to complement [STAR CANTO 3A](http://www.ais.gov.hk/HK_AIP/AIP/AD/HK_AD2-98%20CANTO.pdf);
- `abbey` preset arrival lengthened to include fix `TD`, because [both STAR ABBEY 3A/2B](http://www.ais.gov.hk/HK_AIP/AIP/AD/HK_AD2-98%20ABBEY.pdf) go to `TD`; 3A then travels the whole distance to `LIMES`, while 2B ends at `TD`;
- `canto` preset arrival shortened to `MURRI`, because that is the last common fix between [STAR CANTO 3A](http://www.ais.gov.hk/HK_AIP/AIP/AD/HK_AD2-98%20CANTO.pdf) and [STAR CANTO 1G](http://www.ais.gov.hk/HK_AIP/AIP/AD/HK_AD2-98G.pdf).

Could the map creator @Fechulo review these changes?

For the same reasons as `canto` arrival, I would also love to also shorten the `betty` arrival to `MANGO`, but unfortunately `MANGO` is outside of the radar. @Maverick283 and @erikquinn, is it possible then to have different preset arrivals with the same first (and maybe even second) fix? As you can see from the [general STAR chart](http://www.ais.gov.hk/HK_AIP/AIP/AD/HK_AD2-87.pdf), ideally we would like to have two sets of arrivals beginning with `BETTY` and `MURRI`. Will the planes crash outside of radar?

**Comment by Fechulo** [comment]
I've tested it and it looks and works well

**Comment by tedrek** [comment]
@siyuanr, It is possible to have multiple arrivals which use the same fixes, but there is the potential for collisions.  In the case of BETTY/MANGO the two arrivals appear to join at BETTY and so will enter the terminal area as a single stream from the players perspective.  I'm also guessing that BETTY 2A and BETTY 2B are assigned based on active runway so generally only one of them would be in use at a time.  Therefore it's simplest to just use a single preset arrival for aircraft from that STAR.

The BETTY arrival can be shortened to MANGO, there is no reason why the final fix needs to be in the terminal area, it's even possible to assign a different heading after the last fix as compared to just overflying it.  However in this case I'd suggest just modeling BETTY 2B due to the wind.

The one time I have created two arrivals which use the same fixes was when I had both turbojets and props coming from the same direction, in that case the props were given a slower speed and altitude so the jets would fly over top thereby avoiding any collision possibility until they entered the terminal area.

**Comment by siyuanr** [comment]
@tedrek Just shortened BETTY arrival to MANGO (cf. #378).

**Comment by siyuanr** [comment]
I now added all the restricted airspaces. Please review #385 and refer to discussion in #384.

**Comment by erikquinn** [comment]
#384 is closed via the merge of #385.

This issue is now resolved. Closing.
